chromatic_fitting.py

```diff
@@ -78,127 +86,277 @@ def summarize_parameters(self):
     def plot_priors(self, n=3):
         """
         Plot n prior samples from the parameter distributions defined by the user
         :parameter n
         Number of priors to plot (default=3)
         """
-        data = self.get_data()
-        prior_predictive_trace = self.sample_prior(ndraws=n)
-        for i in range(n):
-            flux_for_this_sample = np.array([prior_predictive_trace[f'wavelength_{w}_data'][i] for w in range(data.nwave)])
-            data.fluxlike[f'prior-predictive-{i}'] = flux_for_this_sample
-        data.imshow_quantities()
+        # setup the models, data and orbits in a format for looping
+        if self.optimization == "separate":
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+            prior_predictive_traces = self.sample_prior(ndraws=n)
+        else:
+            datas = [self.get_data()]
+            prior_predictive_traces = [self.sample_prior(ndraws=n)]
+
+        for nm, (data, prior_predictive_trace) in enumerate(zip(datas, prior_predictive_traces)):
+            for i in range(n):
+                flux_for_this_sample = np.array(
+                    [prior_predictive_trace[f'wavelength_{w+nm}_data'][i] for w in range(data.nwave)])
+                data.fluxlike[f'prior-predictive-{i}'] = flux_for_this_sample
+            data.imshow_quantities()
 
-    def plot_posteriors(self, trace, n=3):
+    def plot_posteriors(self, n=3):
         """
         Plot n posterior samples from the parameter distributions defined by the user
         :parameter trace
         PyMC3 trace object (need to run sample first)
         :parameter n
         Number of posteriors to plot (default=3)
         """
-        data = self.get_data()
-        posterior_predictive_trace = self.sample_posterior(trace, ndraws=n)
-        for i in range(n):
-            flux_for_this_sample = np.array(
-                [posterior_predictive_trace[f'wavelength_{w}_data'][i] for w in range(data.nwave)])
-            data.fluxlike[f'posterior-predictive-{i}'] = flux_for_this_sample
-        data.imshow_quantities()
+        if self.optimization == "separate":
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+            # traces = self.trace
+            posterior_predictive_traces = self.sample_posterior(n)
+        else:
+            datas = [self.get_data()]
+            # traces = [self.trace]
+            posterior_predictive_traces = [self.sample_posterior(n)]
+
+        for nm, (data, posterior_predictive_trace) in enumerate(zip(datas, posterior_predictive_traces)):
+            for i in range(n):
+                flux_for_this_sample = np.array(
+                    [posterior_predictive_trace[f'wavelength_{w+nm}_data'][i] for w in range(data.nwave)])
+                data.fluxlike[f'posterior-predictive-{i}'] = flux_for_this_sample
+            data.imshow_quantities()
+
+    def extract_mean_posteriors(self,summary, i):
+        # there's definitely a sleeker way to do this
```

Comment:
Ha, probably, but if it works!?

It'd be good to have an easy way to spit out a summary of any particular fit containing parameters used (fixed or fitted) with a summary for their uncertainty (like central 68% confidence interval or maybe hdi_16% to hdi_84% or some such)?

Reply:
It'd also be good to have a quick "make transmission spectrum table" function that summarizes wavelength-dependent quantities and their uncertainties, with wavelength as the first column?

Reply:
Added a first-attempt function to do this:
![Screenshot 2022-06-17 at 17 20 45](https://user-images.githubusercontent.com/34891147/174411918-4292f4cb-4774-482d-ac1f-7bb737c6a1e3.png)

Reply:
But so far it relies on the uncertainty you provide being in the summary table, which you'd have to specify in advance